src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/DlsFlsFilterLeafReader.java

```diff
@@ -106,23 +106,25 @@
     private final MaskedFieldsMap maskedFieldsMap;
     private final ShardId shardId;
     private final boolean maskFields;
+    private final Salt salt;
 
     private DlsGetEvaluator dge = null;
 
 
     DlsFlsFilterLeafReader(final LeafReader delegate, final Set<String> includesExcludes,
                            final Query dlsQuery, final IndexService indexService, final ThreadContext threadContext,
                            final ClusterService clusterService,
-                           final AuditLog auditlog, final Set<String> maskedFields, final ShardId shardId) {
+                           final AuditLog auditlog, final Set<String> maskedFields, final ShardId shardId, final Salt salt) {
         super(delegate);
 
-        maskFields = (auditlog.getComplianceConfig().isEnabled() && maskedFields != null && maskedFields.size() > 0);
+        maskFields = (maskedFields != null && maskedFields.size() > 0);
```

Comment:
Is the similar check on line 1099 necessary? If it is removed here, should it be also removed in `getRuntimeMaskedFieldInfo()`?